nixos/modules/tasks/network-interfaces.nix

```diff
@@ -503,6 +528,23 @@ in
               loaded with <literal>ovs-ofctl</literal> within one atomic operation.
             '';
           };
+          # TODO: provide list of openflow protocols and check that openFlowVersion is included in supportedOpenFlowVersions
+          supportedOpenFlowVersions = mkOption {
+            type = types.listOf types.str;
+            example = [ "OpenFlow10" "OpenFlow13" "OpenFlow14" ];
+            default = [ "OpenFlow13" ];
+            description = ''
+              Supported versions to enable on this switch.
+            '';
+          };
+
+          # openFlowVersion = mkOption {
```

Comment:
This option should be then deprecated in `nixos/modules/rename.nix`

Reply:
`OpenFlow13` is a good default?

Reply:
`OpenFlow13` is a good default because it's the most commonly supported OpenFlow version among the community (also in most hardware switches).

About the option, I don't know if we should remove it or not actually. `ovs-ofctl` supports having a list of protocols as arguments, which is why I originally removed the `openFlowVersion`, but they are not strictly the same:
- `openFlowVersion` could be the version used with ofctl
- `supportedOpenFlowVersions` is the list of protocols supported by the vswitch

Of course, using `supportedOpenFlowVersions` for both works, but the user won't be able to decide which version is used to push the rules. It is not clear how the protocol negotiation works, but probably it will take the highest compatible version.

I think the use case where you'd want to specify a `openFlowVersion` is limited, also because higher OpenFlow version tries to keep retro-compatibility (at least after 1.3).

A compromise would be that this option defaults `supportedOpenFlowVersions` list. But I don't know if you can reference other config option in a config option...

I agree that the option should be properly deprecated if we actually drop it (I didn't specifically know about this before).

Reply:
Ok. If you keep this option please update the description so people understand the difference between the two.

Reply:
I settled for having both options. At the moment, they are a list of strings, but it could be improved by having the versions values as attributes.